client/with_dry_runs.go

```diff
@@ -21,133 +21,117 @@ import (
 	"strings"
 	"time"
 
-	"github.com/ethereum/go-ethereum"
-	"github.com/ethereum/go-ethereum/accounts/abi"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/rpc"
 	"github.com/mysteriumnetwork/payments/bindings"
 	"github.com/pkg/errors"
 )
 
+type estimatable interface {
+	GetGasLimit() uint64
+	toEstimator(ethClientGetter) (*bindings.ContractEstimator, error)
+	toEstimateOps() *bindings.EstimateOpts
+}
+
+// ErrorTransactionReverted when contract execution is interrupted with an error./
+type ErrorTransactionReverted struct {
+	Err    rpc.Error
+	Reason string
+}
+
+func (e ErrorTransactionReverted) Error() string {
+	return e.Err.Error()
```

Comment:
Are we hiding the Reason field for a reason? I think we could produce a better error here

Reply:
It's already wrapped in error text. And `Reason` is message extracted from error.